### ISSUE TITLE
assign does not respect type of target field symbol

### ISSUE THREAD

**Issue by arcanist123** [abaplint/transpiler#417]
This code should produce a runtime error (string is not char4). It is running instead 
``` abap
TYPES:
  BEGIN OF ts_test,
    a TYPE c LENGTH 4,
  END OF ts_test.
  DATA ls_test type ts_test.


  FIELD-SYMBOLS <lv> TYPE string.
  ASSIGN COMPONENT 'A' OF STRUCTURE ls_test TO <lv>.
  WRITE <lv>.
```

**Comment by arcanist123** [comment]
I was hesitant if this is a duplicate of https://github.com/abaplint/transpiler/issues/404 .Still decided to open another one, since there it is a simple casting, and the assumption is that the type of pointer and the underlying field are matching
